agentMET4FOF/agents.py

```diff
@@ -456,18 +456,30 @@ def send_plot(self, fig=plt.Figure()):
         fig : Figure
             Can be either matplotlib figure or plotly figure
 
+        mode : str
+            "image" - converts into image via encoding at base64 string.
+            "plotly" - converts into plotly figure using `mpl_to_plotly`
+            Default: "image"
+
         Returns
         -------
         The message format is {'from':agent_name, 'plot': data, 'senderType': agent_class}.
         """
+
         if isinstance(fig, matplotlib.figure.Figure):
-            #graph = self._convert_to_plotly(fig) #unreliable
-            graph = self._fig_to_uri(fig)
+            if mode == "plotly":
+                graph = self._convert_to_plotly(fig)
+            elif mode == "image":
+                graph = self._fig_to_uri(fig)
         elif isinstance(fig, dict): #nested
-            for key in fig.keys():
-                fig[key] = self._fig_to_uri(fig[key])
+            if mode == "plotly":
+                for key in fig.keys():
+                    fig[key] = self._convert_to_plotly(fig[key])
+            if mode == "image":
+                for key in fig.keys():
+                    fig[key] = self._fig_to_uri(fig[key])
             graph = fig
-        else:
+        else: #a plotly figure
```

Comment:
Before we expicitly asked for the values of `mode` which could have been replaced with an `else` statement whenever we asked the second time for being equal to something (`plotly` in this case). Here we assume automatically it is a plotly figure if not the other one. I think we should decide here explicitly as well or do the input checks in the beginning as suggested earlier. If we do the input checks, this bit here could stay as it is.